modules/metastore/modules/metastore_search/metastore_search.module

```diff
@@ -1,58 +1,66 @@
 <?php
 
 use Drupal\Core\Entity\EntityInterface;
+use Drupal\metastore\NodeWrapper\Data;
 
 /**
- * Manage item tracking of search api items.
+ * Determine whether the supplied entity is a dataset.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $entity
+ *   The entity in question.
+ *
+ * @return bool
+ *   Whether the entity is a dataset.
  */
-function metastore_search_search_api_tracking(EntityInterface $entity, string $operation) {
-  if ($entity->getEntityTypeId() != 'node') {
-    return;
-  }
-
-  if ($entity->bundle() != 'data') {
-    return;
-  }
-
-  $wrapper = new Drupal\metastore\NodeWrapper\Data($entity);
-  if ($wrapper->getDataType() != 'dataset') {
-    return;
-  }
-
-  $acceptableOperation = ['Inserted', 'Updated', 'Deleted'];
-
-  if (!in_array($operation, $acceptableOperation)) {
-    return;
-  }
-
-  $method = "trackItems{$operation}";
-
-  $storage = \Drupal::service('entity_type.manager')
-    ->getStorage('search_api_index');
-
-  /* @var $index \Drupal\search_api\Entity\Index */
-  $index = $storage->load('dkan');
-
-  $index->{$method}('dkan_dataset', [$entity->uuid()]);
+function _is_dataset(EntityInterface $entity): bool {
```

Comment:
I think it's best not to have a function name this generic in the global namespace... we could easily end up defining a function with the same name in any other dkan module. can we rename to `_dkan_metastore_search_is_dataset()`?

Reply:
Fair enough.